src/utils/detect-server-settings.mjs

```diff
@@ -364,4 +364,38 @@ const formatSettingsArrForInquirer = function (frameworks) {
   return formattedArr.flat()
 }
 
+/**
+ * Returns a copy of the provided config with any plugins provided by the
+ * server settings
+ * @param {*} config
+ * @param {Partial<import('./types').ServerSettings>} settings
+ * @returns {*} Modified config
+ */
+export const getConfigWithPlugins = (config, settings) => {
```

Comment:
This logic used to exist in `src/commands/dev/dev.mjs` but it makes more sense next to the rest of the server settings code.